pathtools/fs.go

```diff
@@ -220,8 +224,16 @@ func (m *mockFs) Exists(name string) (bool, bool, error) {
 }
 
 func (m *mockFs) IsDir(name string) (bool, error) {
+	origName := name
 	name = filepath.Clean(name)
 	name = m.followSymlinks(name)
+
+	dir := filepath.Dir(name)
+	dir = m.followSymlinks(dir)
+	if _, ok := m.files[dir]; ok {
```

Comment:
Doesn't this check need to happen all the way up the tree?

Reply:
I'll try making IsDir recursive.